lib/python/qmk/cli/new/keymap.py

```diff
@@ -0,0 +1,65 @@
+"""
+This script automates the copying of the default keymap into your own keymap.
+"""
+import os
+import sys
+
+from milc import cli
+from qmk.path import keymap, check_directories, create_keymap_directory
+
+
+# TODO(orngshdw): add debugging code if needed:
+#   @cli.argument('--debug', dest='debug', action='store_true', help='specify to turn on debugging messages')
+@cli.entrypoint('Create a new keymap.c from the default keyboard of your choosing')
+def main(cli):
+
+    # get keyboard
+    keyboard = input("Keyboard Name: ")
+    # get username
+    username = input("New Username: ")
+
+    # generate keymap paths
+    kb_path = os.path.abspath(keymap(keyboard))
+    print(kb_path)
+    keymap_path_default = kb_path + "/default"
+    keymap_path = kb_path + "/" + username
+
+    # check directories
+    check_directories(kb_path, keymap_path_default)
+    check_directories(keymap_path, exist=False)
+
+    # create user directory with default keymap files
+    create_keymap_directory(keymap_path_default, keymap_path)
+
+    # end message to user
+    print("%s keymap directory created in: %s\n" % (
+        username,
+        keymap_path,
+    ))
+    print("Compile a firmware file with your new keymap by typing: \n")
+    print("make %s:%s\n" % (
+        keyboard,
+        username,
+    ))
+    print("from the qmk_firmware directory")
+
+
+# TODO (orngshdw): delete before merging to master
+#   for reference until we decide if we need logging/debugging for this script
```

Comment:
FYI milc provides extensive logging and argument parsing already. Use `cli.log` to access the pre-configured [`logging`](https://docs.python.org/3.5/library/logging.html) objects. My next project is to revamp the configuration engine, but in the meantime you can set command line flags and access them via `cli.config.general` or `cli.args`, depending on whether it's a flag (--arg or -arg) or a positional argument.

I know this is probably holdover from the standalone script, but I wanted to provide that context until the full documentation is written.